test/unit/dom/dom.js

```diff
@@ -222,4 +222,105 @@ suite('DOM', function() {
       };
     });
   });
+
+  suite('p5.prototype.createSlider', function() {
+    var myp5;
+
+    setup(function(done) {
+      new p5(function(p) {
+        p.setup = function() {
+          myp5 = p;
+          done();
+        };
+      });
+    });
+
+    teardown(function() {
+      myp5.remove();
+    });
+
+    var elt;
+
+    teardown(function() {
+      if (elt && elt.parentNode) {
+        elt.parentNode.removeChild(elt);
+        elt = null;
+      }
+    });
+
+    test('should be instance of p5.Element', () => {
+      expect(myp5.createSlider(5, 10) instanceof p5.Element).to.eql(true);
+    });
+
+    test('should create a slider', function() {
+      let slider = myp5.createSlider(5, 10, 8, 1);
+      elt = document.createElement('input');
+      elt.type = 'range';
+      elt.min = 5;
+      elt.max = 10;
+      elt.step = 1;
+      elt.value = 8;
+      expect(JSON.stringify(slider.elt)).to.eql(JSON.stringify(elt));
+    });
+  });
+
+  suite('p5.prototype.createA', function() {
+    var myp5;
+
+    setup(function(done) {
+      new p5(function(p) {
+        p.setup = function() {
+          myp5 = p;
+          done();
+        };
+      });
+    });
+
+    teardown(function() {
+      myp5.remove();
+    });
+
+    var elt;
+
+    teardown(function() {
+      if (elt && elt.parentNode) {
+        elt.parentNode.removeChild(elt);
+        elt = null;
+      }
+    });
+
+    test('should be instance of p5.Element', () => {
+      expect(
+        myp5.createA('http://p5js.org/', 'this is a link') instanceof p5.Element
+      ).to.eql(true);
+    });
+
+    test('should create anchor tag', function() {
+      let anchor = myp5.createA('http://p5js.org/', 'p5');
+      elt = document.createElement('a');
+      elt.href = 'http://p5js.org/';
+      elt.innerHTML = 'p5';
+      expect(JSON.stringify(anchor.elt)).to.eql(JSON.stringify(elt));
+    });
+
+    testSketchWithPromise('anchor tag is hovered', function(
+      sketch,
+      resolve,
+      reject
+    ) {
+      sketch.setup = function() {
+        var elem = sketch.createA('http://p5js.org/', 'this is a link');
+        elem.mouseMoved();
```

Comment:
I'm not sure I understand this test, can you explain?
it looks like the `testSketchWithPromise` is used correctly in any case

Reply:
I was confused as it was used a bit differently [here](https://github.com/processing/p5.js/blob/master/test/unit/dom/dom.js#L172) as there was no then block used for resolution.